gui/layers/_markers_layer.py

```diff
@@ -279,18 +309,23 @@ def _set_selected_markers(self, indices):
         indices : sequence of int
             Indices to check if marker at.
         """
-        in_slice_markers, matches = self._slice_markers(indices)
+        in_slice_markers, matches, scale = self._slice_markers(indices)
 
         # Display markers if there are any in this slice
         if len(in_slice_markers) > 0:
             shape = (len(in_slice_markers), 2)
             index_array = np.broadcast_to(indices[:2], shape)
             distances = abs(in_slice_markers - index_array)
             # Get the marker sizes
-            if isinstance(self.size, (list, np.ndarray)):
-                sizes = self.size[matches]
+            if isinstance(self.size, np.ndarray):
+                if len(self.size) == len(indices):
+                    sizes = (self.size[0]+self.size[1])/2*scale
+                elif isinstance(self.size, np.ndarray) and len(self.size.shape)==2:
+                    sizes = self.size[matches,:2].mean(axis=1)*scale
+                else:
+                    sizes = self.size[matches]*scale
```

Comment:
The above suggestion might help this case also... Converting `self.size` to a canonical format early can greatly simplify the logic everywhere else.

Reply:
I def get what you mean with converting `self.size` to canonical format on input - my hesitation was two-fold
(1) I don't know if vispy markers routine is slower when taking a size array vs a size int, and so didn't want to force an array when one wasn't needed. This may be completely untrue or negligible. It is also easy to test, and I will do so. 
(2) i was just worried about the behavior when you say do `layer.size = 10` in the notebook and then go `layer.size` and get back some massive array. Is this bad user experience / design - or is this what is appropriate in this case? It will definitely make the internal code much simpler.
Would love your feedback @jni

Reply:
Good points!

(1) I suspect it will not be a big difference, but looking forward to hearing about your checks,

(2) This is indeed more concerning. Perhaps we can have `.size` (what the user passes in), and `._size` (our canonical version which is used everywhere internally). Would that work?

Reply:
Right now there is a `._size` as the thing that gets returned when `.size` is called, using a the property decorator. How about we make a `._size_array` that gets computed whenever `.size` is set, and then we can just use the `._size_array` for these computations assuming (1) is fine.

Reply:
If we are using properties already, we can do whatever with the internal format. I suggest `.size` gets stored into `._size_original` or somesuch, and this is what is returned with `.size`, and `._size` is the canonical format. Otherwise the code might get a bit cumbersome. But I don't have a strong opinion about the naming here, just a suggestion.

Reply:
yup, i get it. will give it a try, thanks!

Reply:
Something that I have noticed before that is quite annoying is that if I can't take advantage of logic inside the setter function on init of the object because calling the setter function has a refresh in it that will lead to all sorts of errors if it is called during construction. We might want to have some form of flag that blocks refreshes till after construction, like a "freeze" or something. For now I will have to duplicate code, but that would be cleaner. Any thoughts from @kne42 on that too?